src/Common.cmake

```diff
@@ -69,8 +69,35 @@ macro(common_project_options)
         CACHE STRING "Fallbacks for the RelWithDebInfo build type")
   endif()
 
-  # Generate compile_commands.json to make it easier to work with clang based tools
-  set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+  if(CMAKE_GENERATOR MATCHES ".*Makefile*." OR CMAKE_GENERATOR MATCHES ".*Ninja*")
+    # Generate compile_commands.json to make it easier to work with clang based tools
+    set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+    # Make a symbol link of compile_commands.json on the source dir to help clang based tools find it
+    if(WIN32)
+      # Detect whether cmake is run as administrator (only administrator can read the LOCAL SERVICE account reg key)
+      cmake_host_system_information(RESULT IS_ADMINISTRATOR QUERY WINDOWS_REGISTRY "HKU\\S-1-5-19")
```

Comment:
Nice job here 👍🏻

Reply:
Though quite an assumption. I can't test whether it works until I return home on February 18.